.github/workflows/android-staging-deploy.yaml

```diff
@@ -0,0 +1,185 @@
+name: Android staging deploy
```

Comment:
It seemed a bit cleaner to me to pull out a separate workflow for staging deploy rather than add more checks for `github.event_name == 'pull_request'`. Some of these common pieces should probably be pulled out into a composable workflow to reduce duplication.